### PR TITLE
[Feature][Flink-SQL-connector] add flink sql connector hbase 1.2&2.2 and docs

### PR DESCRIPTION
**Pull request by legendtkl**
## Purpose of this pull request
issue ref: https://github.com/apache/incubator-seatunnel/issues/1930

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Seem like you should add license in known-dependencies.txt

**Comment by legendtkl** [comment]
PR to be UPDATED.

**Comment by legendtkl** [comment]
Hi, @ruanwenjun  and @BenJFan , I'm afraid the git workflow might need some improvement.

I tried the failed command in my machine (MacOS), it works.

I think the failure might caused by maven concurrency building '-T'
```bash
./mvnw -T 2C -B clean verify -Dmaven.test.skip=false -Dcheckstyle.skip=true -Dscalastyle.skip=true -Dlicense.skipAddThirdParty=true --no-snapshot-updates
```

I suggest to modify it with no concurrency thread build.

What do you thinks?

File: seatunnel-connectors/seatunnel-connectors-flink-sql/flink-sql-connector-hbase-2.2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>flink-sql-connector-hbase-2.2</artifactId>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-connector-hbase-2.2_${scala.binary.version}</artifactId>
+      <version>${flink.version}</version>
+    </dependency>
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connectors-flink-sql/flink-sql-connector-hbase-2.2/pom.xml]: the version should statement in root pom(dependency-management)

**Comment by CalvinKirs** [comment]
> Hi, @ruanwenjun and @BenJFan , I'm afraid the git workflow might need some improvement.
> 
> I tried the failed command in my machine (MacOS), it works.
> 
> I think the failure might caused by maven concurrency building '-T'
> 
> ```shell
> ./mvnw -T 2C -B clean verify -Dmaven.test.skip=false -Dcheckstyle.skip=true -Dscalastyle.skip=true -Dlicense.skipAddThirdParty=true --no-snapshot-updates
> ```
> 
> I suggest to modify it with no concurrency thread build.
> 
> What do you thinks?

What do you seem to be missing? This CI will list the newly added dependencies,

**Comment by ruanwenjun** [comment]
> Hi, @ruanwenjun and @BenJFan , I'm afraid the git workflow might need some improvement.
> 
> I tried the failed command in my machine (MacOS), it works.
> 
> I think the failure might caused by maven concurrency building '-T'
> 
> ```shell
> ./mvnw -T 2C -B clean verify -Dmaven.test.skip=false -Dcheckstyle.skip=true -Dscalastyle.skip=true -Dlicense.skipAddThirdParty=true --no-snapshot-updates
> ```
> 
> I suggest to modify it with no concurrency thread build.
> 
> What do you thinks?

Yeah, if you may get error if you use -T 2C to build in your laptop. But I don't think we need to remove the concurrency setting, since this is used for GitHub CI, you can just remove this when you build on your laptop.

**Comment by legendtkl** [comment]
> > Hi, @ruanwenjun and @BenJFan , I'm afraid the git workflow might need some improvement.
> > I tried the failed command in my machine (MacOS), it works.
> > I think the failure might caused by maven concurrency building '-T'
> > ```shell
> > ./mvnw -T 2C -B clean verify -Dmaven.test.skip=false -Dcheckstyle.skip=true -Dscalastyle.skip=true -Dlicense.skipAddThirdParty=true --no-snapshot-updates
> > ```
> >   
> > I suggest to modify it with no concurrency thread build.
> > What do you thinks?
> 
> What do you seem to be missing? This CI will list the newly added dependencies,

UT failed with this message "Error:  Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single (bin) on project seatunnel-dist: Execution bin of goal org.apache.maven.plugins:maven-assembly-plugin:3.3.0:single failed: basedir D:\a\incubator-seatunnel\incubator-seatunnel\seatunnel-dist\..\seatunnel-connectors\seatunnel-connectors-flink-sql-dist\target\lib does not exist -> [Help 1]"

Just as I comment above, the maven concurrency build with `-T C2` might cause this problem. I run the same command on my laptop, it works.

To verify this, I submit a new commit to remove the `-T C2` in git worklow, now the step in CI passed. 

So I think we need to update the Github CI workflow (already present in this PR) to avoid the misjudgement. Or other improvement of github CI works is ok.

cc @CalvinKirs  @ruanwenjun

**Comment by Hisoka-X** [comment]
@ruanwenjun @CalvinKirs Do this PR can merge?

**Comment by legendtkl** [comment]
> @ruanwenjun @CalvinKirs Do this PR can merge?

Hold it now. cc @BenJFan

File: tools/dependencies/known-dependencies.txt [hunk 2]
```diff
@@ -232,36 +236,50 @@ hadoop-yarn-common-2.7.7.jar
...
+hbase-protocol-shaded-2.2.3.jar
 hbase-replication-2.0.0.jar
 hbase-replication-2.1.0.jar
 hbase-server-2.0.0.jar
 hbase-server-2.1.0.jar
 hbase-shaded-miscellaneous-2.1.0.jar
+hbase-shaded-miscellaneous-2.2.1.jar
 hbase-shaded-netty-2.1.0.jar
+hbase-shaded-netty-2.2.1.jar
 hbase-shaded-protobuf-2.1.0.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies.txt:281]: Cool,  but some NOTICE and License may be missing. Another question, can we unify the version, (maybe not), for example we have multiple versions of hbase.

**Comment by EricJoy2048** [comment]
Hi, what's news about this pr?

**Comment by EricJoy2048** [comment]
This pr is closed because it has not responded for too long. If necessary, please reopen it

**Pull request closed without merging**